i18n/extract.py

```diff
@@ -179,6 +186,21 @@ def run(self, args):
         self.rename_source_file('django-saved.po', 'django.po')
         self.rename_source_file('djangojs-saved.po', 'djangojs.po')
 
+        # Collect a list of full paths to the files to remove POT-Creation-Date from
+        path_names_to_clean_metadata = [self.source_msgs_dir.joinpath(filename) for filename in files_to_clean.union({
```

Comment:
Is there a reason to do this as a union with the partials after the "Cleaning" section instead of just adding the partials to `files_to_clean` and cleaning them with the other files?

I'd think we could add the creation date removal step to the "Cleaning" loop and avoid opening file with `polib` twice.

Something like:
```py
remove_pot_creation_date(pofile):
    """
    Remove POT-Creation-Date
    """
    pofile.metadata.pop('POT-Creation-Date', None)
```
then
```py
        # Finish each file.
        for filename in files_to_clean:
            LOG.info('Cleaning %s', filename)
            pofile = polib.pofile(self.source_msgs_dir.joinpath(filename))
            # replace default headers with edX headers
            fix_header(pofile)
            # replace default metadata with edX metadata
            fix_metadata(pofile)
            # remove key strings which belong in messages.po
            strip_key_strings(pofile)
            # remove the POT-Creation-Date
            remove_pot_creation_date(pofile)
            pofile.save()
```

Reply:
I couldn't find a reason that prevents us from combining the mechanism as you suggested @brian-smith-tcril . Except that, if we do, then we must remove the `PO-Revision-Date` [metadata](https://github.com/openedx/i18n-tools/blob/723a0fbc3158afe9d0c42f4d5ed2cc6315ce480c/i18n/extract.py#L237) too. Are we OK with that?

if we keep `PO-Revision-Date`, it will always be added with a new datetime, and we'll be back to the original problem

cc: @OmarIthawi

Reply:
setting the dates to fictional dates such as `1970-01-01` would be an acceptable alternative in my opinion. re: code formatting and refactoring I have no prefernce.

Reply:
I don't see any benefit to having a fictional date in there over just removing the `PO-Revision-Date` entirely. Is there something I'm overlooking?

Reply:
Done, I had to refactor the code into new method because pylint (50 statements quality-limit)